### PR TITLE
Feature/add stock options functionality

### PR DESCRIPTION
**Pull request by CompoTypo**
This branch contains a small refactor for cookie and crumb acquisition as well as the primary feature of a `getStockOptions` method.

I have been using this package for quite a while now and love it. I have been getting more into options and I would like to enhance my own personal trading info infra by adding onto this library. 

I tried my best to stick to the conventions laid out in the API/decoder and followed the formatting and checks as described in the CONTRIB. Hopefully it is up to standard.

If you have any recommendations or questions, please don't hesitate to comment.

Best,
Compo

### CODE REVIEW

File: src/ResultDecoder.php [hunk 1]
```diff
@@ -282,4 +300,36 @@ public function transformQuotesSumamary(string $responseBody): array
...
+            throw new ApiException('Yahoo Search API returned an invalid result.', ApiException::INVALID_RESPONSE);
+        }
+
+        $results = $decoded['optionChain']['result'];
+
+        // Single element is returned directly in "quote"
+        return array_map(function (array $item) {
+            return $this->createOption($item);
+        }, $results);
+    }
```
Comment (scheb) [inline src/ResultDecoder.php]: I'd like to have a test case similar to `transformQuotes_jsonGiven_createArrayOfQuote` for this method. So we can ensure all expected fields are properly mapped.

**Review by scheb — COMMENTED** [review]
Look really good! I'd just want to have a test case for the value mapping and then we'd be good to merge.

File: src/ResultDecoder.php (hunk 1 above)
Comment (CompoTypo) [reply src/ResultDecoder.php]: PR is updated with additional tests for the options result

**Comment by CompoTypo** [comment]
Thank you for the positive words! I do recommend re-reviewing the work since the models are expanded upon greatly since the initial pr. Thank you for your patience. Please let me know if there are any new issues.

**Comment by scheb** [comment]
Wow, you really did a great effort with the unit tests, thank you!

- Would you please rename the integration tests to match the naming pattern as described here? -> 
https://github.com/scheb/yahoo-finance-api/blob/4.x/CONTRIBUTING.md#creating-a-pull-request
- Psalm has some complaints (see build), would you please handle those?
- Adding an example call to the README.md would be great 👍 

Other than that, I'd be happy to merge that in.

**Comment by CompoTypo** [comment]
I squashed another commit on top. The changes should take care of those few requests. Have a great weekend!

**Comment by scheb** [comment]
Thank you for your work. Has been released under v4.10.0

**Pull request merged**